IdentityCore/src/controllers/MSIDRequestControllerFactory.m

```diff
@@ -101,6 +130,86 @@ @implementation MSIDRequestControllerFactory
+{
+    // Nested auth protocol - Reverse client id & redirect uri
+    if ([parameters isNestedAuthProtocol])
+    {
+        [parameters reverseNestedAuthParametersIfNeeded];
+    }
+    
+    MSIDSilentController *fallbackController = nil;
+    
+    if ([parameters shouldUseBroker])
+    {
+        if (parameters.allowUsingLocalCachedRtWhenSsoExtFailed)
+        {
+            fallbackController = [[MSIDSilentController alloc] initWithRequestParameters:parameters
+                                                                         forceRefresh:YES
+                                                                 tokenRequestProvider:tokenRequestProvider
+                                                                                error:error];
+            fallbackController.isLocalFallbackMode = YES;
+        }
+    
+        MSIDSilentController *xpcController = nil;
+#if TARGET_OS_OSX
+        if (parameters.msidXpcMode != MSIDXpcModeDisable && [MSIDXpcSilentTokenRequestController canPerformRequest])
+        {
+            xpcController = [[MSIDXpcSilentTokenRequestController alloc] initWithRequestParameters:parameters
+                                                                                           forceRefresh:forceRefresh
+                                                                                   tokenRequestProvider:tokenRequestProvider
+                                                                          fallbackInteractiveController:fallbackController
+                                                                                                  error:error];
+            if (parameters.msidXpcMode == MSIDXpcModeFull || parameters.msidXpcMode == MSIDXpcModeOverride)
+            {
+                // If in Xpc full mode, the XPCController will work as a isolated controller when SsoExtension cannotPerformRequest
+                fallbackController = xpcController;
+                xpcController = nil;
+            }
+        }
+#endif
+        
+        BOOL shouldSkipSsoExtension = parameters.msidXpcMode == MSIDXpcModeOverride;
+        
+        if (!shouldSkipSsoExtension && [MSIDSSOExtensionSilentTokenRequestController canPerformRequest])
+        {
+            fallbackController = [[MSIDSSOExtensionSilentTokenRequestController alloc] initWithRequestParameters:parameters
+                                                                                                    forceRefresh:forceRefresh
+                                                                                            tokenRequestProvider:tokenRequestProvider
+                                                                                   fallbackInteractiveController:xpcController?:fallbackController
+                                                                                                           error:error];
+        }
+    }
+    
```

Comment:
Can we log when fallbackController is nil?

Reply:
added